src/main/java/input_feed/distributed/JSONToDistributedR1CS.java

```diff
@@ -0,0 +1,196 @@
+package input_feed.distributed;
+
+import algebra.fields.AbstractFieldElementExpanded;
+import configuration.Configuration;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import relations.objects.Assignment;
+import relations.objects.LinearTerm;
+import relations.objects.R1CSConstraintsRDD;
+import relations.r1cs.R1CSRelationRDD;
+import scala.Tuple2;
+
+import java.io.FileReader;
+import java.util.ArrayList;
+
+public class JSONToDistributedR1CS<FieldT extends AbstractFieldElementExpanded<FieldT>> {
+    private final String filePath;
+    private final int numInputs;
+    private final int numAuxiliary;
+    private final int numConstraints;
+    private final FieldT fieldParameters;
+
+    public JSONToDistributedR1CS(final String _filePath, final FieldT _fieldParameters) {
+        filePath = _filePath;
+        fieldParameters = _fieldParameters;
+
+        JSONParser parser = new JSONParser();
+        JSONObject jsonObject;
+        JSONArray header = new JSONArray();
+
+        try {
+            Object obj = parser.parse(new FileReader(filePath));
```

Comment:
You are parsing this file 3 times. Once in the constructor, then in loadR1CS and again in loadWitness. Given this json could be quite large, should we just keep the parsed version in memory?